drivers/disk/ramdisk.c

```diff
@@ -33,6 +37,21 @@ static int disk_ram_access_status(struct disk_info *disk)
 
 static int disk_ram_access_init(struct disk_info *disk)
 {
+#if IS_ENABLED(CONFIG_DISK_RAM_EXTERNAL)
+	mm_reg_t virt_start;
```

Comment:
Can we reduce the lifetime and visibility of this variable by moving its declaration under the if statement below?
looks like it produces warnings of unused variable in case when we have `CONFIG_DISK_RAM_EXTERNAL` enabled and `DEVICE_MMIO_IS_IN_RAM` is disabled

Reply:
Done